boskos/ranch/storage.go

```diff
@@ -415,7 +415,9 @@ func (s *Storage) syncDynamicResourceLifeCycles(newDRLCByType, existingDRLCByTyp
 	for _, existingDRLC := range existingDRLCByType {
 		newDRLC, existsInNew := newDRLCByType[existingDRLC.Name]
 		if existsInNew {
-			if !reflect.DeepEqual(existingDRLC, newDRLC) {
+			if !reflect.DeepEqual(existingDRLC.Spec, newDRLC.Spec) {
```

Comment:
I am not 100% happy with this as this may surprise us down the line should we ever add a Status field. Alternate would be to copy `ObjectMeta` from `existingDLRC` onto `newDLRC`, but that seems just as bad, just different.

Reply:
The alternate-alternate would be to copy the spec from new into existing, then add the existing object to the list to update. I'm not sure which of these is best (or least worst).

Reply:
I don't especially like how the current version loses any other metadata that may have been added to the object.

Reply:
Updated to just copy ObjectMeta